.gitlab-ci.yml

```diff
@@ -1,34 +1,186 @@
 
-.global_testing_win: &global_testing_win
+.global_env_linux: &global_env_linux
   script:
-    - Invoke-Expression $Env:ARTENOLIS_SOFT_PATH"\julia\"$Env:JULIA_VER"\bin\julia --color=yes --project=@. -e 'import Pkg; Pkg.test(; coverage = true)'"
+    - $ARTENOLIS_SOFT_PATH/julia/$JULIA_VER/bin/julia --inline=yes --check-bounds=yes --color=yes --project=@. -e 'import Pkg; Pkg.test(; coverage = true)'
 
-linux julia v1.6:
-  stage: build
-  variables:
-    JULIA_VER: "v1.6.0"
-  <<: *global_settings
-  <<: *global_testing
+.global_env_win: &global_env_win
+  script:
+    - $global:LASTEXITCODE = 0 # Note the global prefix.
+    - Invoke-Expression $Env:ARTENOLIS_SOFT_PATH"\julia\"$Env:JULIA_VER"\bin\julia --inline=yes --check-bounds=yes --color=yes --project=@. -e 'import Pkg; Pkg.test(; coverage = true)'"
+    - exit $LASTEXITCODE
 
-windows10:
-  stage: build
+.global_env_win10: &global_env_win10
   tags:
     - windows10
-  variables:
-    JULIA_VER: "v1.6.0"
-  <<: *global_testing_win
+  <<: *global_env_win
+
+.global_env_mac: &global_env_mac
+  tags:
+    - mac
+  script:
+    - $ARTENOLIS_SOFT_PATH/julia/$JULIA_VER/Contents/Resources/julia/bin/julia --inline=yes --check-bounds=yes --color=yes --project=@. -e 'import Pkg; Pkg.test(; coverage = true)'
+
+.global_build_apptainer: &global_build_apptainer
+  image:
+    name: "quay.io/singularity/singularity:$APPTAINER_DOCKER_TAG"
+    # the image entrypoint is the singularity binary by default
+    entrypoint: ["/bin/sh", "-c"]
+  tags:
+    - privileged
+
+#
+# TESTS
+#
+# The "basic" required test that gets triggered for the basic testing, runs in
+# any available docker and current julia
+#
+
+docker:julia1.7:
+  stage: test
+  image: $CI_REGISTRY/r3/docker/julia-custom
+  script:
+    - julia --check-bounds=yes --inline=yes --project=@. -e "import Pkg; Pkg.test(; coverage = true)"
+  after_script:
+    - julia --project=test/coverage test/coverage/coverage-summary.jl
+  <<: *global_trigger_pull_request
+
+#
+# The required compatibility test to pass on branches&tags before the docs get
+# built & deployed
+#
+
+linux:julia1.7:
+  stage: test
+  tags:
+    - slave01
+  <<: *global_trigger_full_tests
+  <<: *global_julia17
+  <<: *global_env_linux
+
+#
+# Additional platform&environment compatibility tests
+#
+
+windows10:julia1.7:
+  stage: test
+  <<: *global_trigger_full_tests
+  <<: *global_julia17
+  <<: *global_env_win10
+
+mac:julia1.7:
+  stage: test
+  <<: *global_trigger_full_tests
+  <<: *global_julia17
+  <<: *global_env_mac
+
+#
+# CONTAINERS
+#
+
+apptainer-test:
+  stage: assets
+  script: |
+    alias apptainer=singularity
+    apptainer build gigasom-test.sif gigasom.def
+  <<: *global_build_apptainer
+  <<: *global_trigger_test_containers
+
+apptainer-release:
+  stage: assets
+  script:
+    - |
+      # build the container
+      alias apptainer=singularity
+      apptainer build gigasom.sif gigasom.def
+    - |
+      # push to GHCR
+      alias apptainer=singularity
```

Comment:
best feature ever

Reply:
I hope they don't remove it in the next version. :D :D